### ISSUE TITLE
Rationalize VB and C# anonymous delegate types.

### ISSUE THREAD

**Issue by CyrusNajmabadi** [dotnet/roslyn#55292]
Right now VB and C# represent anonymous delegates in different ways.  Both are INamedTypes with TYpeKind.Delegate.  However, a VB anonymous delegate reports IsAnonymousType=true, while a C# one reports CanBeReferencedByName=false.

We should investigate and determine if it is really appropriate for these two languages to deviate here in the representation.  If there is a good reason, then that's fine and the split can remain.  However, if there isn't, we should unify to help higher layers have a consistent way of handling both of these uniformly.

**Comment by RikkiGibson** [comment]
I was confused by the issue description. Isn't `IsAnonymousType=true` consistent with `CanBeReferencedByName=false`? Do the respective symbols give different answers for the same API?

**Comment by CyrusNajmabadi** [comment]
Yes.  The answers for vb and c# are different. C# does not report 'true' for 'isanoymoustype'
